pkg/helpers/motd/motd_test.go

```diff
@@ -0,0 +1,51 @@
+package motd
+
+import (
+	"bytes"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func configmap(namespace, name, key, content string) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			key: content,
+		},
+	}
+}
+
+func TestDisplayMOTD(t *testing.T) {
+	var tests = []struct {
+		description string
+		expected    string
+		obj         runtime.Object
+	}{
+		{"Correctly displays message", "This is a legal notice\n", configmap("openshift-config", "motd", "message", "This is a legal notice")},
+		{"No message because of malformed notice (misspelled key)", "", configmap("openshift-config", "motd", "mesage", "This is a legal notice")},
+		{"No message because of misconfigured notice (wrong name)", "", configmap("openshift-config", "bad-name", "message", "This is a legal notice")},
+		{"No message because notice not found in namespace", "", configmap("default", "motd", "mesage", "This is a legal notice")},
```

Comment:
The key should probably be called "message" in the last test, because the purpose of the test is that the configmap is not loaded from the default namespace, not that the key is misspelled. It's not a big deal, because the code fails where it's expected to fail, it's made me go "hmm?" for a bit.

Reply:
Thanks! will fix.